x/pep/keeper/msg_server_request_private_identity.go

```diff
@@ -0,0 +1,157 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	sdkerrors "cosmossdk.io/errors"
+	commontypes "github.com/Fairblock/fairyring/x/common/types"
+	kstypes "github.com/Fairblock/fairyring/x/keyshare/types"
+	"github.com/Fairblock/fairyring/x/pep/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	channeltypes "github.com/cosmos/ibc-go/v8/modules/core/04-channel/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (k msgServer) RequestPrivateIdentity(goCtx context.Context, msg *types.MsgRequestPrivateIdentity) (*types.MsgRequestPrivateIdentityResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	reqID, found := k.GetRequestId(ctx, msg.Creator, msg.ReqId)
+	if found || len(reqID.ReqId) != 0 {
+		return nil, types.ErrReqIDAlreadyExists
+	}
+
+	k.SetRequestId(ctx, types.RequestId{
```

Comment:
Do we need a separate request ID KVStore for the encrypted id ? Right now is sharing the same KVStore with general keyshare request id

Reply:
i am using a separate store key for private requests. This particular check basically checks over ALL keyshare requests (both private and general).

I don't want to have two different entries (one as a general keyshare, and one as a private keyshare) with the same ID